src/posthog-core.js

```diff
@@ -617,7 +622,9 @@ PostHogLib.prototype.capture = addOptOutCheckPostHogLib(function (event_name, pr
     const has_unique_traits = options !== __NOOPTIONS
 
     if (this.get_config('request_batching') && (!has_unique_traits || options._batchKey) && !options.send_instantly) {
-        data['timestamp'] = new Date()
```

Comment:
Actually I think we could pull this out of this `request_batching` branch and make it `data['timestamp'] = options.timestamp || new Date()` – this would be more straightforward.